client/packages/system/src/InventoryAdjustmentReason/Components/InventoryAdjustmentReasonSearchInput.tsx

```diff
@@ -12,43 +13,43 @@ import {
   InventoryAdjustmentReasonRowFragment,
 } from '../api';
 
-export enum Adjustment {
-  Reduction,
-  Addition,
-  None,
-}
-
 interface InventoryAdjustmentReasonSearchInputProps {
   value: InventoryAdjustmentReasonRowFragment | null;
   width?: number | string;
   onChange: (
     inventoryAdjustmentReason: InventoryAdjustmentReasonRowFragment | null
   ) => void;
   autoFocus?: boolean;
-  adjustment: Adjustment;
+  adjustmentType: AdjustmentTypeInput;
   isError?: boolean;
 }
 
 export const InventoryAdjustmentReasonSearchInput: FC<
   InventoryAdjustmentReasonSearchInputProps
-> = ({ value, width, onChange, autoFocus = false, adjustment, isError }) => {
+> = ({
+  value,
+  width,
+  onChange,
+  autoFocus = false,
+  adjustmentType,
+  isError,
+}) => {
   const { data, isLoading } =
     useInventoryAdjustmentReason.document.listAllActive();
-  const isRequired = data?.totalCount !== 0;
-  const isDisabled = adjustment === Adjustment.None || !isRequired;
```

Comment:
Hm, if we're going to remove the `None` option, we need another mechanism to disable this input.

From stocktake, before counted packs is touched, or when snapshot packs === counted packs, reason input should be disabled. With this change, now can enter a reason:

![Screenshot 2024-04-15 at 3 13 16 PM](https://github.com/msupply-foundation/open-msupply/assets/55115239/37a8c7ea-9f1c-4e2f-8bfb-7dedec025a02)

And if you submit with the reason, you get an error... 

Perhaps can just add an `isDisabled` prop, reuse the logic from the old `getAdjustment` in StocktakeLineEditTables?

Reply:
Right, good catch. Have added the `isDisabled` prop and set it to true in StocktakeLineEditModal when the `countedNumberOfPacks` is `null` (that's the initial value). 91affa621

That enough you think?

Reply:
Above it looks like it also used to be disabled when `countedNumberOfPacks` === `snapshotNumberOfPacks`, and when `countThisLine` is `false`. I'd keep consistent with current logic 👍

Reply:
Cool, I missed that bit above. Thanks.

d1bf85e41